workflows/fastquorum.nf

```diff
@@ -41,19 +41,41 @@ workflow FASTQUORUM {
     ch_versions = Channel.empty()
     ch_multiqc_files = Channel.empty()
 
+    // TODO: This won't work for single-ends!
+    ch_samplesheet
+        .branch {
+            meta, fastqs ->
+                single  : fastqs.size() == 2
```

Comment:
I must be missing something, since where do we group _across_ samples to gather the FASTQs?  The sample sheet has one row per sample, with fastq1 and fastq2 only, so it's not possible to have `fastqs.size() > 2` from one row in the sample sheet.

See: https://github.com/nf-core/rnaseq/blob/b89fac32650aacc86fcda9ee77e00612a1d77066/workflows/rnaseq.nf#L213-L234

Reply:
See also: https://github.com/nf-core/sarek/blob/b5b766d3b4ac89864f2fa07441cdc8844e70a79e/workflows/sarek/main.nf#L247-L254

Reply:
Another one to take inspiration from: https://github.com/nf-core/rnasplice/blob/1d0494ae3402d1a46e0adadad24f81a0ff855c77/workflows/rnasplice.nf#L161-L181

Reply:
The `PIPELINE_INITIALISATION` [subworkflow](https://github.com/nf-core/fastquorum/blob/c321fb3156c3033ddfda45226fe39e0f3a31e14d/subworkflows/local/utils_nfcore_fastquorum_pipeline/main.nf#L92-L110 ) includes the following code:
```
    Channel
        .fromSamplesheet("input")
        .map {
            meta, fastq_1, fastq_2 ->
                if (!fastq_2) {
                    return [ meta.id, meta + [ single_end:true ], [ fastq_1 ] ]
                } else {
                    return [ meta.id, meta + [ single_end:false ], [ fastq_1, fastq_2 ] ]
                }
        }
        .groupTuple()
        .map {
            validateInputSamplesheet(it)
        }
        .map {
            meta, fastqs ->
                return [ meta, fastqs.flatten() ]
        }
        .set { ch_samplesheet }
```
The output after the `validateInputSamplesheet` step looks like this: 
```
[
  [id:sample1, read_structure:5M2S+T 5M2S+T, single_end:false], 
  [ 
    [sample1_L001_R1.fq.gz, sample1_L001_R2.fq.gz], 
    [sample1_L002_R1.fq.gz, sample1_L002_R1.fq.gz]
  ]
]
  
[
  [id:sample2, read_structure:5M2S+T 5M2S+T, single_end:false], 
  [
    [sample1_L001_R1.fq.gz, sample1_L001_R2.fq.gz]
  ]
]
```
The `groupTuple` collected all rows that have exactly the same meta and grouped the remaining output files together. The flatten on the next line was then flattening across both levels of the nested fastqs:
```
[
  [id:sample1, read_structure:5M2S+T 5M2S+T, single_end:false], 
  [sample1_L001_R1.fq.gz, sample1_L001_R2.fq.gz, sample1_L002_R1.fq.gz, sample1_L002_R1.fq.gz]
]
  
[
  [id:sample2, read_structure:5M2S+T 5M2S+T, single_end:false], 
  [sample1_L001_R1.fq.gz, sample1_L001_R2.fq.gz]
]
```
This flatten is new in the `PIPELINE_INITIALISATION` subworkflow introduced by the template update, and was throwing me off (and I just wanted to get something to work so I could use it in my work, where I have fastq files sequenced across multiple lanes!).

Having removed that `flatten`, the typical code will now work, as I've pushed now.